applications/asset_tracker/src/main.c

```diff
@@ -2095,13 +2152,54 @@ static void lte_handler(const struct lte_lc_evt *const evt)
 	case LTE_LC_EVT_CELL_UPDATE:
 		LOG_INF("LTE cell changed: Cell ID: %d, Tracking area: %d",
 			evt->cell.id, evt->cell.tac);
-#if defined(CONFIG_NRF_CLOUD_CELL_POS)
+#if defined(CONFIG_NRF_CLOUD_CELL_POS) && !defined(CONFIG_CELL_POS_MULTICELL)
 		/* Request location based on new network info */
 		if (data_send_enabled() && evt->cell.id != -1) {
 			k_work_reschedule_for_queue(&application_work_q,
 						       &send_cell_pos_request_work,
 						       K_SECONDS(1));
 		}
+#endif
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS:
+		LOG_INF("Neighbor cell measurements received: %d",
+			evt->cells_info.ncells_count);
+#if defined(CONFIG_CELL_POS_MULTICELL)
+		/* change the LED color based on the number of neighbors being used */
+		if (evt->cells_info.ncells_count > 1) {
+			ui_led_set_pattern(UI_LED_GPS_FIX);
+		} else if (evt->cells_info.ncells_count) {
+			ui_led_set_pattern(UI_LED_GPS_SEARCHING);
+		} else {
+			ui_led_set_pattern(UI_LTE_CONNECTED);
+		}
+		if (evt->cells_info.current_cell.id == 0) {
+			LOG_DBG("Cell ID not valid.");
+			break;
+		}
+
+		/* Copy current cell information. */
+		memcpy(&cell_data.current_cell,
+		       &evt->cells_info.current_cell,
+		       sizeof(struct lte_lc_cell));
+
+		/* Copy neighbor cell information if present. */
+		if (evt->cells_info.ncells_count > 0 && evt->cells_info.neighbor_cells) {
+			memcpy(neighbor_cells,
+			       evt->cells_info.neighbor_cells,
+			       sizeof(struct lte_lc_ncell) * evt->cells_info.ncells_count);
+
+			cell_data.ncells_count = evt->cells_info.ncells_count;
+		} else {
+			cell_data.ncells_count = 0;
+		}
+
+		/* Request location based on new network info */
+		if (data_send_enabled() && evt->cells_info.current_cell.id != -1) {
```

Comment:
should the check for invalid cell id be done at the top of this case?

Reply:
I was going to ask about that.  Your code from the multicell sample checks for 0 (which I preserved), but the single cell case right above (LTE_LC_EVT_CELL_UPDATE) checks for -1.  I'm not sure where the -1 comes from, or if it could occur for multicell.

Reply:
Went ahead and combined checks for 0 and -1 with <= 0.

Reply:
the `!= 0` check was based on the comment for `lte_lc_cells_info`:
```
/** @brief Structure containing results of neighbor cell measurements.
 *	   The current cell information is valid if the current cell ID is non-zero.
 *	   The ncells_count member indicates whether or not the structure contains
 *	   valid neighbor cell information. If it is zero, no cells were found, and
 *	   the information in the rest of structure members do not contain valid data.
 */
```

but `<= 0` seems better in general?

Reply:
Actually, no.  Turns out `id` in this struct and the related one for single cell are uint32_t.  I've asked for clarification by the modem team.